### PR TITLE
[feature](pipelineX) use expected<T, Status> in local_state

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

File: be/src/pipeline/pipeline_x/operator.h [hunk 1]
```diff
@@ -21,25 +21,25 @@
...
-#define CREATE_LOCAL_STATE_RETURN_NULL_IF_ERROR(local_state) \
-    auto _sptr = state->get_local_state(id());               \
-    if (!_sptr) return nullptr;                              \
-    auto& local_state = _sptr->template cast<LocalState>();
-
-#define CREATE_SINK_LOCAL_STATE_RETURN_NULL_IF_ERROR(local_state) \
-    auto _sptr = state->get_sink_local_state(id());               \
-    if (!_sptr) return nullptr;                                   \
-    auto& local_state = _sptr->template cast<LocalState>();
+#define CREATE_LOCAL_STATE_RETURN_STATUS_IF_ERROR(local_state) \
```
Comment (JackDrogon) [inline be/src/pipeline/pipeline_x/operator.h:24]: why use macro，it's better to use inline function not macro. inline function is good for debug and can get a clean namespace. macro is only for define scope or create an anonymous variable

**Review by JackDrogon — CHANGES_REQUESTED** [review]

File: be/src/pipeline/pipeline_x/operator.h (hunk 1 above)
Comment (yiguolei) [reply be/src/pipeline/pipeline_x/operator.h:24]: could  not agree more

Comment (Mryange) [reply be/src/pipeline/pipeline_x/operator.h:24]: But this is used to create the local_state variable.

**Pull request closed without merging**